R/agg_scale.R

```diff
@@ -260,150 +260,22 @@ agg <- function(dt,
 
   # Do aggregation ----------------------------------------------------------
 
-  if (col_type == "interval") {
-    cols <- paste0(col_stem, "_", c("start", "end"))
-
-    dt_intervals <- unique(dt[, .SD, .SDcols = cols])
-    overlapping_dt <- identify_overlapping_intervals(dt_intervals)
-
-    if (nrow(overlapping_dt) > 0) {
-      if (!quiet) message("Collapsing ", col_stem, " to the most detailed common set of intervals")
-      # collapse entire dataset to most detailed common set of intervals for the
-      # {col_stem} variable being aggregated over
-      dt <- collapse_common_intervals(
-        dt = dt[, .SD, .SDcols = c(id_cols, value_cols)],
-        id_cols = id_cols,
-        value_cols = value_cols,
-        col_stem = col_stem,
-        agg_function = agg_function,
-        missing_dt_severity = missing_dt_severity,
-        overlapping_dt_severity = overlapping_dt_severity,
-        include_missing = TRUE
-      )
```

Comment:
This was slowing down interval aggregation before, trying to collapse all the input data intervals to a common set of intervals. Specifically the internal checks for missing or overlapping intervals.